packages/ui/components/ui/slider/index.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import * as SliderPrimitive from '@radix-ui/react-slider';
+import { cn } from '../../../lib/utils';
+import { Thumbs } from './thumbs';
+
+interface BaseProps {
+  /**
+   * Note that this is an array. You can pass more than one value to have
+   * multiple draggable points on the slider.
+   */
+  value: number[];
+  onValueChange: (value: number[]) => void;
+  /**
+   * If set to `true`, will show a tooltip with the current value represented by
+   * the thumb that the user's mouse is over. If set to a function, will call
+   * that function with the current value, then use the returned string as the
+   * tooltip content.
+   */
+  thumbTooltip?: boolean | ((value: number) => string);
+}
+
+// When `segmented` is `true`, the other props are required.
+interface PropsWithSegmented extends BaseProps {
+  /** Segment the slider at each step */
+  segmented: true;
+  min: number;
+  max: number;
+  /** The step size to count by */
+  step: number;
+}
+
+interface PropsWithoutSegmented extends BaseProps {
+  /** Segment the slider at each step */
+  segmented?: false;
+  min?: number;
+  max?: number;
+  /** The step size to count by */
+  step?: number;
+}
+
+/**
+ * Renders a draggable range slider:
+ * |---o-------------|
+ *
+ * @example
+ * ```tsx
+ * <Slider
+ *   min={0}
+ *   max={10}
+ *   step={2}
+ *   value={value}
+ *   onValueChange={onValueChange}
+ *   segmented // to show gaps in the slider for each option
+ *   thumbTooltip={value => `Current value: ${value}`}
+ * />
+ * ```
+ */
+const Slider = ({
+  segmented,
+  value,
+  onValueChange,
+  min,
+  max,
+  step,
+  thumbTooltip,
+}: PropsWithSegmented | PropsWithoutSegmented) => {
+  const segmentCount =
+    !!segmented && (max - min + 1) % step === 0 ? Math.max((max - min + 1) / step - 1, 1) : 1;
```

Comment:
For future maintainability, can we somehow extract this out and/or annotate what this is doing?

Reply:
Ah, yeah, big oversight on my part 😆 I extracted this logic to a verbose `getSegmentCount` helper [here](https://github.com/penumbra-zone/web/pull/1279/files#diff-09f5d1ff494b1453dec33b98e4f5dad528c265e744fd7ce3341139e5ae164874R42-R74).